spec/0.3-draft/spec.md

```diff
@@ -0,0 +1,1121 @@
+
+JSON schema: [rect-node.json](core/rect-node.json)
+
+
+## Oval
+An oval is a visual node extension, to define the visual appearance of a node as an oval.
+An oval in a square bounding box is a circle.
+
+An oval has the exact same properties as a [Rectangle](#rectangle), just the rendering is different.
+The oval shall be rendered as an ellipse, within the bounding box defined by the node's position and size.
+
+
+JSON schema: [oval-node.json](core/oval-node.json)
+
+
+## Arrow
+An arrow is a visual node that connects two point coordinates.
+It should be rendered as a straight line, with optional direction markers at the start and end.
+
+
+| Property      | JSON Type | OCIF Type | Required     | Contents                | Default   |
+|---------------|-----------|-----------|--------------|-------------------------|-----------|
+| `strokeWidth` | `number`  | number    | optional     | The line width.         | `1`       |
+| `strokeColor` | `string`  | string    | optional     | The color of the arrow. | `#FFFFFF` |
+| `start`       | `array`   | number    | **required** | The start point.        | n/a       |
+| `end`         | `array`   | number    | **required** | The end point.          | n/a       |
+| `startMarker` | `string`  | string    | optional     | Marker at the start.    | `none`    |
+| `endMarker`   | `string`  | string    | optional     | Marker at the end.      | `none`    |
+
+- **strokeWidth**:
+  The line width in logical pixels. Default is `1`. Inspired from SVG `stroke-width`.
+ 
+- **strokeColor**:
+  The color of the arrow. Default is white (`#FFFFFF`). Inspired from SVG `stroke`.
+ 
+- **start**:
+  The start point of the arrow. The array contains the x and y coordinates.  
+  The z-coordinate, if present, is used only in 3D canvas apps.
+ 
+- **end**:
+  The end point of the arrow. The array contains the x and y coordinates.
+  The z-coordinate, if present, is used only in 3D canvas apps.
+
+- **startMarker**:
+  The marker at the start of the arrow. 
+  Directions from the perspective of the arrow. 
+  Possible values are:
+    - `none`: No special marker at the start. A flat line end at the start.
+    - `incoming`: An incoming arrow at the start. Indicating flow from start point into the arrow start.
+    - `outgoing`: An outgoing arrow at the start. Indicating flow from the arrow end into the end point.
+
+- **endMarker**:
+  The marker at the end of the arrow. 
+  Directions from the perspective of the arrow. 
+  Possible values are:
+    - `none`: No special marker at the end. A flat line end at the end.
+    - `incoming`: An incoming arrow at the end. Indicating flow from the arrow end into the end point.
+    - `outgoing`: An outgoing arrow at the end. Indicating flow from the start point into the arrow start.
```

Comment:
Not a blocker, just an observation: it seems that we are deciding that the "core" representation of an arrow has three types of "endings": arrowhead, "incoming", and none. These implicitly represent "from" and "to" relationships, which is kinda nice, but I think we should also represent start and end nodes explicitly. (see other comment).

Probably worth a review of different canvases to see what start and end markers are commonly supported and what their names are.

Reply:
Simplify to `none` and `arrowhead`, perhaps.

Reply:
simplified to 'none' or 'arrowhead', as discussed